README.md

```diff
@@ -0,0 +1,138 @@
+
+### Suyama cofactor test
+The Suyama test utilizes Fermat’s little theorem to test a Fermat number's cofactor for probable primality.
+It was devised by Hiromi Suyama in 1984 and subsequently improved upon by Hendrik W. Lenstra, Jr.
+If a Fermat number $F_n$ is composite and equal to $P \times C$, where $P$ is the product of known factors and $C$ is the cofactor, then the Suyama test is performed as follows:
+
+* Calculate $A = b^{F_n-1}$ (mod $F_n$) for some base $b$. If base = 3 then this is the square mod $F_n$ of the final Pepin residue.
+* Calculate $B = b^{P-1}$ (mod $F_n$).
+* Calculate $R = (A - B)$ (mod $C$). If $R = 0$ then the cofactor $C$ is a probable prime; otherwise it is composite.
+
+Proving the cofactor composite requires finding only one base for which $R \neq 0$. cofact (main branch) only supports base $b = 3$ for the Suyama test. So far, this has been sufficient since each Fermat cofactor from $F_{12}$ through $F_{30}$ is currently composite.
+
+Finally, the cofactor $C$ can be tested to determine if it is a prime power by calculating the greatest common divisor $G = \text{gcd}(A - B, C)$. If $G = 1$ then the cofactor is not a prime power. If $G \neq 1$ then the cofactor is a prime power and is divisible by $G$.
+
+### Computation
+The less performance-critical calculations in cofact use the GNU Multiple Precision (GMP) arithmetic library. However for the heavy lifting of modular squarings required by Pépin's test, the `gwnum` library is used since it is multi-threaded and therefore provides much higher performance. Even so, cofact (main branch) does not have Gerbitz error checking of calculations and does not save interim results to a file. So, for a Fermat cofactor test that is expected to run more than a few hours, it is preferable to first either generate the proof file using mprime / Prime95 (which has Gerbitz error checking and checkpoint / restart capability) or download the proof file from Catherine's [website](https://64ordle.au/fermat/). Once a Fermat number's proof file is in hand, `cofact -upr` can be used to test the new cofactor whenever a new factor of the Fermat number is discovered.
+
+### Proof files
+mprime / Prime95 is capable of generating a verifiable delay function proof file for any Fermat number through $F_{30}$. Proof files have been generated and collected for $F_{14}$ through $F_{29}$ and are available from Catherine’s [website](https://64ordle.au/fermat/). If you wish to use mprime / Prime95 to generate a proof file for $F_{30}$ (requires a system with AVX512 support), we would be [most interested](https://www.mersenneforum.org/showthread.php?t=22668&page=6) in knowing about it!
```

Comment:
You might want to change the link to Mersenne forum, which now probably ought to point to: https://www.mersenneforum.org/node/17112/page3

Reply:
The old link appears to still work for now, but may stop working in the future. So I changed to the new "node" format, but left off page3 which, if the thread grows, would be somewhere in the middle of the thread.